testing/trino-product-tests/src/main/java/io/trino/tests/product/hudi/TestHudiMinioReads.java

```diff
@@ -0,0 +1,240 @@
+
+        onHudi().executeQuery("INSERT INTO default." + tableName + " PARTITION (dt, hh) SELECT 1 AS id, 'a1' AS name, 1000 AS ts, '2021-12-09' AS dt, '10' AS hh");
+        onHudi().executeQuery("INSERT INTO default." + tableName + " PARTITION (dt = '2021-12-09', hh='11') SELECT 2, 'a2', 1000");
+
+        List<QueryAssert.Row> expectedRows = ImmutableList.of(
+                row(1, "a1", 1000, "2021-12-09", "10"),
+                row(2, "a2", 1000, "2021-12-09", "11"));
+
+        try {
+            assertThat(onHudi().executeQuery("SELECT id, name, ts, dt, hh FROM default." + tableName))
+                    .containsOnly(expectedRows);
+            assertThat(onTrino().executeQuery("SELECT id, name, ts, dt, hh FROM hudi.default." + tableName))
+                    .containsOnly(expectedRows);
+
+            expectedRows = ImmutableList.of(row(2, "a2", 1000));
+            assertThat(onTrino().executeQuery("SELECT id, name, ts FROM hudi.default." + tableName + " WHERE dt = '2021-12-09' AND hh = '11'"))
+                    .containsOnly(expectedRows);
+        }
+        finally {
+            onHudi().executeQuery("DROP TABLE default." + tableName);
+        }
+    }
+
+    @Test(groups = {HUDI, PROFILE_SPECIFIC_TESTS})
+    public void testCopyOnWriteTableSelectAfterUpdate()
+    {
+        String tableName = "test_hudi_demo_" + randomTableSuffix();
+        String tableDirectory = "hudi-compatibility-test-" + tableName;
+
+        // create a COW non-partitioned table, with default primaryKey 'uuid' and without preCombineField provided
+        createNonPartitionedHudiTable(tableName, tableDirectory, bucketName);
+
+        List<QueryAssert.Row> expectedRows = ImmutableList.of(
+                row(1, "Trino"),
+                row(2, "rocks"));
+
+        try {
+            assertThat(onHudi().executeQuery("SELECT uuid, col FROM default." + tableName))
+                    .containsOnly(expectedRows);
+            assertThat(onTrino().executeQuery("SELECT uuid, col FROM hudi.default." + tableName))
+                    .containsOnly(expectedRows);
+        }
+        finally {
+            onHudi().executeQuery("DROP TABLE default." + tableName);
+        }
+
+        // update the table
+        onHudi().executeQuery("UPDATE default." + tableName + " SET col = 'Hudi' WHERE uuid = 2");
+
+        expectedRows = ImmutableList.of(
+                row(1, "Trino"),
+                row(2, "Hudi"));
+
+        try {
+            assertThat(onHudi().executeQuery("SELECT uuid, col FROM default." + tableName))
+                    .containsOnly(expectedRows);
+            assertThat(onTrino().executeQuery("SELECT uuid, col FROM hudi.default." + tableName))
+                    .containsOnly(expectedRows);
+        }
+        finally {
+            onHudi().executeQuery("DROP TABLE default." + tableName);
+        }
+    }
+
+    @Test(groups = {HUDI, PROFILE_SPECIFIC_TESTS})
+    public void testMergeOnReadTableSelect()
```

Comment:
MOR table doesn't support partitioned tables? I'm wondering why there's only one test for MOR.

Reply:
Added a test for MOR partitioned table.